### PR TITLE
Example feature matrix

### PR DESCRIPTION
**Pull request by kvark**
This PR introduces a feature matrix to show what feature can be found in which example. I expect we'll be linking to it a lot :)

It also renames the `describe` example so that we can distinguish easily between ones that are framework-based and standalone ones (starting with "hello").

Concerns:
- the star may be a bit too bleak... I haven't found a better one that would be short to type (choosing a long one means editing is uncomfortable).
- sometimes hard to figure out what deserves a feature in the table and what not... still a win to have at least something here
- we need to do something with `capture`, see #387, but I haven't figured out what exactly

also r? @rukai

### CODE REVIEW

**Comment by cwfitzgerald** [comment]
That's interesting, on my machine (windows/firefox) it's quite prominent:

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - `hello-compute` for pure computing
+
+Notably, `capture` example show rendering without a surface/window. It reads back the contents and saves them to a file.
+
+All framework-based examples render to the window.
+
+## Feature matrix
+| Feature                | boids  | cube   | mipmap | msaa-line | shadow | skybox | texture-arrays | water  |
+| ---------------------- | ------ | ------ | ------ | --------- | ------ | ------ | -------------- | ------ |
+| vertex attribute       | :star: | :star: | :star: | :star:    | :star: |        | :star:         | :star: |
```
Comment (cwfitzgerald) [inline examples/README.md:15]: Even though the features themselves are generally not pluralized, I would prefer to see these pluralized as it reads much better in docs.

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+| lines and points       |        |        |        | :star:    |        |        |                |        |
+| sampled color texture  | :star: | :star: | :star: |           |        | :star: | :star:         | :star: |
+| storage texture        | :star: |        |        |           |        |        |                |        |
+| binding array          |        |        |        |           |        |        | :star:         |        |
+| comparison sampler     |        |        |        |           | :star: |        |                |        |
+| subresource view       |        |        | :star: |           | :star: |        |                |        |
+| cubemap                |        |        |        |           |        | :star: |                |        |
+| multisampling          |        |        |        | :star:    |        |        |                |        |
+| off-screen rendering   |        |        |        |           | :star: |        |                | :star: |
+| stencil test           |        |        |        |           |        |        |                |        |
```
Comment (cwfitzgerald) [inline examples/README.md:26]: I'd prefer: `stencil testing`

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+| depth test             |        |        |        |           | :star: |        |                | :star: |
+| read-only depth        |        |        |        |           |        |        |                | :star: |
```
Comment (cwfitzgerald) [inline examples/README.md:28]: depth/stencil? or is it too large?

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+| blending               |        |        |        |           |        |        |                | :star: |
+| render bundle          |        |        |        | :star:    |        |        |                | :star: |
+| compute pass           | :star: |        |        |           |        |        |                |        |
+| optional extension     |        |        |        |           |        |        | :star:         |        |
```
Comment (cwfitzgerald) [inline examples/README.md:32]: do we actually want to list the optional things it supports? We could have something like
`| - binding indexing |` to show it's a sub item of the optional extensions.

File: examples/README.md (hunk 1 above)
Comment (kvark) [reply examples/README.md]: yeah, I unconsciously pluralized a few things anyway :)

File: examples/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+| cubemaps               |        |        |        |           |        | :star: |                |        |
+| multisampling          |        |        |        | :star:    |        |        |                |        |
+| off-screen rendering   |        |        |        |           | :star: |        |                | :star: |
+| stencil testing        |        |        |        |           |        |        |                |        |
+| depth testing          |        |        |        |           | :star: |        |                | :star: |
+| read-only depth        |        |        |        |           |        |        |                | :star: |
+| blending               |        |        |        |           |        |        |                | :star: |
+| render bundles         |        |        |        | :star:    |        |        |                | :star: |
+| compute passes         | :star: |        |        |           |        |        |                |        |
+| optional extensions    |        |        |        |           |        |        |                |        |
```
Comment (cwfitzgerald) [inline examples/README.md:32]: This row should probably also have a star.

Comment (kvark) [reply examples/README.md]: I was more thinking in line of blanking out the row completely. But a star works too :)

**Pull request merged**